thorlcr/activities/loop/thloopslave.cpp

```diff
@@ -1245,10 +1246,12 @@ class CGraphLoopResultReadSlaveActivity : public CSlaveActivity
         if ((int)sequence >= 0)
         {
             assertex(container.queryResultsGraph());
+            Owned<CGraphBase> graph;
             graph_id resultGraphId = container.queryXGMML().getPropInt("att[@name=\"_graphId\"]/@value");
-            if (!resultGraphId)
-                resultGraphId = container.queryResultsGraph()->queryGraphId();
-            Owned<CGraphBase> graph = queryJobChannel().getGraph(resultGraphId);
+            if (resultGraphId == container.queryResultsGraph()->queryGraphId())
+                graph.set(container.queryResultsGraph());
+            else
+                graph.setown(queryJobChannel().getGraph(resultGraphId));
```

Comment:
Does this cope with nested parallel child queries.  Do you need a recursive querResultsGraph(id)?  (I'm not sure of the answer.)